### PR TITLE
Fix last uses of loadProject(fs) and remove deprecated api

### LINKED ISSUE #2013: sdk tests migration to `loadProject(repo)
## Problem 

The inlang SDK tests are blocking the deprecation of the `nodeishfFs` argument in `loadProject` in favor of `repo` 

For context,project ids yielded the hard requirement for the SDK to directly depend on a repo instance. We (@martin-lysk, @janfjohannes, @samuelstroschein) are confident that future requirements will also have a hard requirement for a repo instance and therefore believe that the inlang SDK depending on lix repos directly is the right decision.  

## Proposal

Replace `nodesihFs` with a repo instance in tests. 

```diff
const project = await loadProject({ 
- nodeishFs, 
+ repo
  projectPath: PATH_TO_INLANG_CONFIG 
})
```

- [ ] refactor tests
- [ ] get rid of conditional logic that handles a potentially undefined repo 
- [ ] remove (now redundant) `loadProject` overloads

### PR DESCRIPTION
**Pull request by jldec**
resolves #2013 

- [x] 1177f06 fix loadProject(fs) call in cli
- [x] a41ac52 remove create-project and ide-extension auto-config  
- [x] 8fc05a5 fix loadProject(fs) call from ide-extension initProject 
- [x] 93409f5 remove loadProject(fs) overload from sdk (breaking change) 
- [x] changesets

Note that a41ac52 and 8fc05a5 will be superseded by #1890 which also gets rid of auto-config in the ide-extension.

### CODE REVIEW

**Comment by jldec** [comment]
🤦 of course, removing the overload [breaks create-project](https://github.com/opral/monorepo/actions/runs/7588957682/job/20672583596?pr=2066#step:8:2818)

**Review by samuelstroschein — APPROVED** [review]

**Review by samuelstroschein — APPROVED** [review]
Looks good!

I changed the changeset versioning from patch to minor as we don't really use patches. 

Simple heuristic: Major if the user needs to pay attention (breaking change, major marketing update). Everything else is minor.

**Review by samuelstroschein — CHANGES_REQUESTED** [review]
The API of a `project.id` should not be potentially `undefined` anymore with this update. A project id should always be generated.

**Comment by jldec** [comment]
@samuelstroschein I'm trying to figure out what changes you are requesting.

loadProject() requires a repo arg now, so it will always attempt to create a project id if it does not exist.

Was your request motivated by the windows vs posix path delimiter issues, or some leftover checks for repo, or do you think there are still cases where the project will come back undefined with this PR?

File: inlang/source-code/sdk/src/loadProject.ts [hunk 1]
```diff
@@ -99,15 +67,7 @@ export async function loadProject(args: {
...
-		// TODO: deprecate
-		fs = args.nodeishFs
-	} else if (args.repo) {
-		fs = args.repo.nodeishFs
-	} else {
-		throw new LoadProjectInvalidArgument(`Repo missing from arguments.`, { argument: "repo" })
-	}
+	const fs = args.repo.nodeishFs
 
 	const nodeishFs = createNodeishFsWithAbsolutePaths({
```
Comment (jldec) [inline inlang/source-code/sdk/src/loadProject.ts:72]: @samuelstroschein I think you're asking to remove the tryCatch on L88.

Comment (samuelstroschein) [reply inlang/source-code/sdk/src/loadProject.ts:72]: nope this one https://github.com/opral/monorepo/issues/2069

Comment (samuelstroschein) [reply inlang/source-code/sdk/src/loadProject.ts:72]: https://discord.com/channels/@me/1196510316717154314/1198999268531380225

Comment (jldec) [reply inlang/source-code/sdk/src/loadProject.ts:72]: aah - ok - 🙏 👀

**Review by samuelstroschein — APPROVED** [review]
OK to merge. we can follow up with subsequent changes

**Comment by jldec** [comment]
ok - for clarification - there are two separate outstanding issues revealed by changing the projectId api and looking at the resulting failing tests.

1.  generateProjectId still returns undefined if it's not a "real" repo. To fix that in this PR we would also have to add new logic to do something different like generate a temporary projectId.

2. loadProject() calls normalize() before calling maybeCreateFirstProjectId(). This does not work as expected in Windows because the nodeishFs.stat can't find the normalized Windows path and thinks the .inlang directory passed in does not exist.

**Pull request merged**